### PR TITLE
[zh-cn]: Update translation of Promise.then()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13733

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 1]
```diff
@@ -25,70 +23,82 @@ then(onFulfilled, onRejected)
 - `onRejected` {{optional_inline}}
   - : 当 Promise 变成拒绝状态（rejected）时调用的{{jsxref("Function", "函数")}}。该函数有一个参数，即拒绝的原因（`rejection reason`）。如果该参数不是函数，则会在内部被替换为一个 "Thrower" 函数 (it throws an error it received as argument)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:24]: please update the description of params.

**Comment by yin1999** [comment]
Please using either "merge commit" or "rebase" only to update the base commit, this may not mixin other file changes.

**Comment by JasonLamv-t** [comment]
> Please using either "merge commit" or "rebase" only to update the base commit, this may not mixin other file changes.

Actually, I use `pull -r` before I `push`, so I think I need to spect the target branch.

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 2]
```diff
@@ -21,74 +19,81 @@ then(onFulfilled, onRejected)
...
-// Promise {[[PromiseStatus]]: "pending", [[PromiseValue]]: undefined}
-// "this gets called after the end of the main stack. the value received and returned is: 33"
-// Promise {[[PromiseStatus]]: "resolved", [[PromiseValue]]: 33}
-```
+立即返回一个新的 {{jsxref("Promise")}} 对象，该对象始终处于待定状态，无论当前 Promise 对象的状态如何。
+
+`onFulfilled` 和 `onRejected` 处理函数之一将被执行，以处理当前 Promise 对象的兑现或拒绝。即使当前 Promise 对象已经敲定，这个调用也总是异步发生的。返回的 Promise 对象（称之为 `p`）的行为取决于处理函数的执行结果，遵循一组特定的规则。如果处理函数：
+
+- 返回一个值：`p` 以该返回值作为其值被兑现。
+- 没有返回任何值：`p` 以 `undefined` 作为其值被兑现。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:46]: I prefer this style, could you update this section?

```suggestion
- 返回一个值：`p` 以该返回值作为其兑现值。
- 没有返回任何值：`p` 以 `undefined` 作为其兑现值。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 3]
```diff
@@ -21,74 +19,81 @@ then(onFulfilled, onRejected)
...
+- 抛出一个错误：`p` 以抛出的错误作为其值被拒绝。
+- 返回一个已兑现的 Promise 对象：`p` 以该 Promise 的值作为其值被兑现。
+- 返回一个已拒绝的 Promise 对象：`p` 以该 Promise 的值作为其值被拒绝。
+- 返回另一个待定的 Promise 对象：`p` 保持待定状态，并在该 Promise 对象被兑现/拒绝后立即以该 Promise 的值作为其值被兑现/拒绝。
 
 ## 描述
 
-由于 `then` 和 {{jsxref("Promise.prototype.catch()")}} 方法都会返回 promise，它们可以被[链式调用](/zh-CN/docs/Web/JavaScript/Guide/Using_promises#Chaining)——这同时也是一种被称为**复合**（_composition）_ 的操作。
+`then()` 方法用于为 Promise 对象的完成（兑现或拒绝）安排回调函数。它是 Promise 的基本方法：[thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 协议要求所有类 Promise 对象都提供一个 `then()` 方法，并且 `catch()` 和 `finally()` 方法都会通过调用该对象的 `then()` 方法来工作。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:54]:
```suggestion
`then()` 方法用于为 Promise 对象的完成（兑现或拒绝）设置回调函数。它是 Promise 的基本方法：[thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 协议要求所有类 Promise 对象都提供一个 `then()` 方法，并且 `catch()` 和 `finally()` 方法都会通过调用该对象的 `then()` 方法来工作。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 4]
```diff
@@ -21,74 +19,81 @@ then(onFulfilled, onRejected)
...
+
+有关 `onRejected` 处理函数的更多信息，请参阅 {{jsxref("Promise/catch", "catch()")}} 参考文档。
+
+`then()` 方法返回一个新的 Promise 对象。如果在同一 Promise 对象上两次调用 `then()` 方法（而不是链式调用），则该 Promise 对象将具有两对处理方法。附加到同一 Promise 对象的所有处理方法总是按照它们添加的顺序调用。此外，每次调用 `then()` 方法返回的两个 Promise 对象开始了独立的链，不会等待彼此的解决。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:58]:
```suggestion
`then()` 方法返回一个新的 Promise 对象。如果在同一 Promise 对象上两次调用 `then()` 方法（而不是链式调用），则该 Promise 对象将具有两对处理方法。附加到同一 Promise 对象的所有处理方法总是按照它们添加的顺序调用。此外，每次调用 `then()` 方法返回的两个 Promise 对象开始了独立的链，不会等待彼此的敲定。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 5]
```diff
@@ -21,74 +19,81 @@ then(onFulfilled, onRejected)
...
+
+沿着 `then()` 链产生的 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象总是会被[解决](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise#resolver_函数)——`onFulfilled` 处理函数永远不会收到 thenable 对象，并且任何由处理函数返回的 thenable 对象在传递给下一个处理函数之前总是被解决。这是因为在构造新的 Promise 对象时，由 `executor` 传递的 `resolve` 和 `reject` 函数被保存下来，当当前 Promise 对象敲定时，相应的函数将被调用，并传递兑现值或拒绝原因。解析逻辑来自 {{jsxref("Promise/Promise", "Promise()")}} 构造函数传递的解析函数。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:60]:
```suggestion
沿着 `then()` 链产生的 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象总是会被[解决](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise#resolver_函数)——`onFulfilled` 处理函数永远不会收到 thenable 对象，并且任何由处理函数返回的 thenable 对象在传递给下一个处理函数之前总是被解决。这是因为在构造新的 Promise 对象时，由 `executor` 传递的 `resolve` 和 `reject` 函数被保存下来，在当前 Promise 对象敲定时，相应的函数将被调用，并传递兑现值或拒绝原因。解决逻辑来自 {{jsxref("Promise/Promise", "Promise()")}} 构造函数传递的解决函数。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 6]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
-// logs, in order:
-// Last Then: oops... didn't bother to instantiate and return a promise in the prior then so the sequence may be a bit surprising
+// 按顺序打印：
+// 最后一个 then：哎呀……之前懒得实例化并返回一个 Promise，所以顺序可能有点令人惊讶
 // foobar
 // foobarbaz
 ```
 
-当一个值只是从一个 `then` 内部返回时，它将等价地返回 `Promise.resolve(<由被调用的处理程序返回的值>)`。
+`then()` 方法返回的值的解决方式与 {{jsxref("Promise.resolve()")}} 相同。这意味着该方法支持 [thenable 对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable)，如果返回值不是一个 Promise 对象，则会自动将其封装在一个 Promise 对象中，然后解决。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:137]:
```suggestion
`then()` 方法返回的值的解决方式与 {{jsxref("Promise.resolve()")}} 相同。这意味着该方法支持 [thenable 对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable)，如果返回值不是一个 Promise 对象，则会隐式地将其包装在 `Promise` 中，然后解决。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 7]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
+    console.error(`onRejected 函数被调用：${error.message}`);
   })
   .then(() => {
-    console.log("I am always called even if the prior then's promise rejects");
+    console.log("即使前面的 Promise 被拒绝，我也总是会被调用");
   });
 ```
 
-你也可以在另一个顶层函数上使用链式去实现基于 Promise API 的函数。
+在其他所有情况下，返回的 Promise 最终都会被兑现。在下面的示例中，即使链中的前一个 Promise 被拒绝，第一个 `then()` 返回一个封装着 `42` 的已兑现的 Promise。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:190]:
```suggestion
在其他所有情况下，返回的 Promise 最终都会被兑现。在下面的示例中，即使链中的前一个 Promise 被拒绝，第一个 `then()` 返回一个包装着 `42` 的已兑现的 Promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 8]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
-  const rfab = Reflect.apply.bind; // (thisArg, fn, thisArg, [...args])
-  const nextTick = (fn, ...args) => (
-    fn !== undefined
-    ? Promise.resolve(args).then(rfab(null, fn, null))
-    : nextTickPromise(),
-    undefined
+// 用一个已解决的 Promise——'resolvedProm' 为例，
+// 函数调用 'resolvedProm.then(...)' 立即返回一个新的 Promise，
+// 但是其中的处理函数 '(value) => {...}' 将会异步调用，正如打印输出所示。
+// 新的 Promise 被赋值给 'thenProm'，
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:272]: as they are not JavaScript string, please use quotation marks

```suggestion
// 用一个已解决的 Promise——“resolvedProm”为例，
// 函数调用“resolvedProm.then(...)”立即返回一个新的 Promise，
// 但是其中的处理器“(value) => {...}”将被异步调用，正如打印输出所示。
// 新的 Promise 被赋值给“thenProm”，
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 9]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
+立即返回一个新的 {{jsxref("Promise")}} 对象，该对象始终处于待定状态，无论当前 Promise 对象的状态如何。
+
+`onFulfilled` 和 `onRejected` 处理函数之一将被执行，以处理当前 Promise 对象的兑现或拒绝。即使当前 Promise 对象已经敲定，这个调用也总是异步发生的。返回的 Promise 对象（称之为 `p`）的行为取决于处理函数的执行结果，遵循一组特定的规则。如果处理函数：
+
+- 返回一个值：`p` 以该返回值作为其兑现值。
+- 没有返回任何值：`p` 以 `undefined` 作为其兑现值。
+- 抛出一个错误：`p` 以抛出的错误作为其值被拒绝。
+- 返回一个已兑现的 Promise 对象：`p` 以该 Promise 的值作为其值被兑现。
+- 返回一个已拒绝的 Promise 对象：`p` 以该 Promise 的值作为其值被拒绝。
+- 返回另一个待定的 Promise 对象：`p` 保持待定状态，并在该 Promise 对象被兑现/拒绝后立即以该 Promise 的值作为其值被兑现/拒绝。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:50]:
```suggestion
- 抛出一个错误：`p` 抛出的错误作为其拒绝值。
- 返回一个已兑现的 Promise 对象：`p` 以该 Promise 的值作为其兑现值。
- 返回一个已拒绝的 Promise 对象：`p` 以该 Promise 的值作为其拒绝值。
- 返回另一个待定的 Promise 对象：`p` 保持待定状态，并在该 Promise 对象被兑现/拒绝后立即以该 Promise 的值作为其兑现/拒绝值。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 10]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
+    - `value`
+      - : Promise 对象的兑现值。
 
-- 返回了一个值，那么 `then` 返回的 Promise 将会成为接受状态，并且将返回的值作为接受状态的回调函数的参数值。
-- 没有返回任何值，那么 `then` 返回的 Promise 将会成为接受状态，并且该接受状态的回调函数的参数值为 `undefined`。
-- 抛出一个错误，那么 `then` 返回的 Promise 将会成为拒绝状态，并且将抛出的错误作为拒绝状态的回调函数的参数值。
-- 返回一个已经是接受状态的 Promise，那么 `then` 返回的 Promise 也会成为接受状态，并且将那个 Promise 的接受状态的回调函数的参数值作为该被返回的 Promise 的接受状态回调函数的参数值。
-- 返回一个已经是拒绝状态的 Promise，那么 `then` 返回的 Promise 也会成为拒绝状态，并且将那个 Promise 的拒绝状态的回调函数的参数值作为该被返回的 Promise 的拒绝状态回调函数的参数值。
-- 返回一个未定状态（`pending`）的 Promise，那么 `then` 返回 Promise 的状态也是未定的，并且它的终态与那个 Promise 的终态相同；同时，它变为终态时调用的回调函数参数与那个 Promise 变为终态时的回调函数的参数是相同的。
+    如果 `onFulfilled` 不是一个函数，则内部会被替换为一个*恒等*函数 (`(x) => x`)，它只是简单地将兑现值向前传递。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:28]:
```suggestion
    如果 `onFulfilled` 不是一个函数，则内部会被替换为一个*恒等*函数（`(x) => x`），它只是简单地将兑现值向前传递。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md [hunk 11]
```diff
@@ -21,248 +19,281 @@ then(onFulfilled, onRejected)
...
-// instantly logging the value of thenProm
-console.log(thenProm);
+    - `reason`
+      - : Promise 对象被拒绝的原因。
 
-// using setTimeout we can postpone the execution of a function to the moment the stack is empty
-setTimeout(() => {
-    console.log(thenProm);
-});
+    如果 `onRejected` 不是一个函数，则内部会被替换为一个*抛出器*函数 (`(x) => { throw x; }`)，它会抛出它收到的拒绝原因。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/then/index.md:37]:
```suggestion
    如果 `onRejected` 不是一个函数，则内部会被替换为一个*抛出器*函数（`(x) => { throw x; }`），它会抛出它收到的拒绝原因。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**